android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -75,16 +66,20 @@ public MapController(Activity mainApp, MapView view, String sceneFilePath) {
         mainApp.getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
         assetManager = mainApp.getAssets();
 
-        // Set up gesture recognizers
-        gestureDetector = new GestureDetector(mainApp, this);
-        scaleGestureDetector = new ScaleGestureDetector(mainApp, this);
-        rotateGestureDetector = new RotateGestureDetector(mainApp, this);
-        shoveGestureDetector = new ShoveGestureDetector(mainApp, this);
-        gestureDetector.setOnDoubleTapListener(this);
+        touchManager = new TouchManager(mainApp);
+        setPanResponder(null);
+        setScaleResponder(null);
+        setRotateResponder(null);
+        setShoveResponder(null);
+
+        touchManager.setSimultaneousDetectionAllowed(Gestures.SHOVE, Gestures.ROTATE, false);
+        touchManager.setSimultaneousDetectionAllowed(Gestures.ROTATE, Gestures.SHOVE, false);
+        touchManager.setSimultaneousDetectionAllowed(Gestures.SHOVE, Gestures.SCALE, false);
```

Comment:
And another thing, `LONG_PRESS` gesture should be disallowed after `SCALE` gesture has started.